src/main/java/Storage.java

```diff
@@ -0,0 +1,84 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Takes care of storing the tasks.
+ */
+public class Storage {
+    private String filePath;
+    public static DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    /**
+     * Constructor for Storage.
+     * @param filePath representing a string for file path
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Method to load tasks from the file.
+     *
+     * @return A task list which has tasks loaded from the file
+     */
+    public List<Task> load() {
+        List<Task> tempList = new ArrayList<>();
+        File file = new File(filePath);
+        try {
+            Scanner fileScanner = new Scanner(file);
+            while (fileScanner.hasNextLine()) {
+                String[] token = fileScanner.nextLine().split("\\|");
+                if (token.length <= 2) {
+                    throw new DukeException("Invalid line");
+                }
+                if (token[0].equals("T")) {
+                    tempList.add(new Todo(token[2], token[1]));
+                } else if (token[0].equals("D")) {
+                    tempList.add(new Deadline(token[2], token[1], LocalDateTime.parse(token[3], dtf)));
+                } else if (token[0].equals("E")) {
+                    tempList.add(new Event(token[2], token[1], token[3]));
+                } else {
+                    throw new DukeException("Invalid task type");
+                }
+            }
+            return tempList;
+        } catch (DukeException de) {
+            System.out.println(de);
+        } catch (IOException ie) {
+            System.out.println("IOException");
+        }
+        return tempList;
+    }
+
+    /**
+     * Method to write to the file.
+     *
+     * @param tasks a list of tasks to be written to the file
+     */
+    public void writeToFile(List<Task> tasks) throws IOException {
+        File file = new File(filePath);
+        PrintWriter pw = new PrintWriter(file);
+        String output = "";
+        for (Task t : tasks) {
+            if (t.getType().equals("T")) {
```

Comment:
The string manipulation to create a formatted string is done in Storage.java; however, it would be more appropriate to include the formatted String as a method "String formatString( )" in Task.java, and override it in the subclasses of Task. Then, call the formatString( ) method in Storage.java instead.